mon-pix/app/authenticators/oidc.js

```diff
@@ -11,51 +11,37 @@ import fetch from 'fetch';
 
 const { host, afterLogoutUri, endSessionEndpoint } = ENV.poleEmploi;
 
-export default class PoleEmploiAuthenticator extends BaseAuthenticator {
+export default class OidcAuthenticator extends BaseAuthenticator {
   @service session;
   @service location;
 
-  async authenticate({ code, redirectUri, state, authenticationKey }) {
-    let request;
+  async authenticate({ code, redirectUri, state, identityProviderName, authenticationKey }) {
+    const request = {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+      },
+    };
     let serverTokenEndpoint;
 
     if (authenticationKey) {
-      serverTokenEndpoint = `${ENV.APP.API_HOST}/api/pole-emploi/users?authentication-key=${authenticationKey}`;
-      request = {
-        method: 'POST',
-        headers: {
-          Accept: 'application/json',
-        },
-      };
+      serverTokenEndpoint = `${ENV.APP.API_HOST}/api/${identityProviderName}/users?authentication-key=${authenticationKey}`;
     } else {
-      serverTokenEndpoint = `${ENV.APP.API_HOST}/api/pole-emploi/token`;
+      serverTokenEndpoint = `${ENV.APP.API_HOST}/api/${identityProviderName}/token`;
       const bodyObject = {
         code,
         redirect_uri: redirectUri,
         state_sent: this.session.data.state,
         state_received: state,
       };
 
-      const body = Object.keys(bodyObject)
+      request.headers['Content-Type'] = 'application/x-www-form-urlencoded';
+      request.body = Object.keys(bodyObject)
         .map((k) => `${k}=${encodeURIComponent(bodyObject[k])}`)
         .join('&');
 
-      request = {
-        method: 'POST',
-        headers: {
-          Accept: 'application/json',
-          'Content-Type': 'application/x-www-form-urlencoded',
-        },
-        body,
-      };
-
-      // We want to authorize users connected to Pix to connect to Pole Emploi as well
-      // in order to link their Pole Emploi account to the Pix one.
       if (this.session.isAuthenticated) {
         request.headers['Authorization'] = `Bearer ${this.session.data.authenticated.access_token}`;
-        // We must ensure to disconnect the Pix user in order for the session service to fire
-        // the authenticationSucceeded event (and thus execute the ApplicationRouteMixin sessionAuthenticated() method).
-        // see: https://github.com/simplabs/ember-simple-auth/blob/92268fdcb9ac3d1c9f7b0abde4923dade7a0cd62/packages/ember-simple-auth/addon/internal-session.js#L95L106
         await this.session.invalidate();
       }
```

Comment:
Remarque : 
Cette ligne est propre à Pôle Emploi mais ne concerne pas la CNAV
`request.headers['Authorization'] = `Bearer ${this.session.data.authenticated.access_token}`;`

Il s'agit du moment où l'on va déconnecter l'utilisateur de Pix pour l'envoyer s'authentifier chez le partenaire.
Coté Pôle Emploi on stocke l'access token car on veut recupérer cet utilisateur précédemment déco pour le réconcilier. 
Coté CNAV ce n'est pas le cas, on veut seulement déconnecter.

Reply:
On pourrait même garder les commentaires pour cette histoire de déconnexion

Reply:
mais pourtant on a teste et cote CNAV on est bien deconnecte 🤷‍♀️

Reply:
C'est parce que le back se charge de réconcilier ou non. 
Ici en réalité on ne fait que stocker l'access token en vu de récupérer le userId pour l'api pour qu'il fasse la réconciliation.

Reply:
On peut donc laisser comme ça mais je pense qu'il serait bon de garder les commentaires non ?

Reply:
ca me va de garder les commentaires, et je pense quand on travaillera sur la deconnexion, peut etre qu'on trouvera un moyen d'ameliorer ca